### PR TITLE
Initial pull request for Samsung Galaxy Ace 4 (SM-G357FZ)

### PR DESCRIPTION
**Pull request by garethppls**
Adds initial support for Samsung Galaxy Ace 4 (SM-G357FZ):

Adds support for:
- Root storage
- Buttons
- Display
- Touchscreen
- SD card 
- WiFi / Bluetooth
-  4G / GPS 
- NFC

### CODE REVIEW

File: arch/arm/boot/dts/Makefile [hunk 1]
```diff
@@ -991,6 +991,7 @@ dtb-$(CONFIG_ARCH_QCOM) += \
 	qcom-msm8660-surf.dtb \
 	qcom-msm8916-samsung-e7.dtb \
 	qcom-msm8916-samsung-fortunaltezt.dtb \
+	qcom-msm8916-samsung-heatqlte.dtb \
```
Comment (TravMurav) [inline arch/arm/boot/dts/Makefile:994]: We usually add all devices to aarch64 then "link" it to 32bit arm. See how qcom-msm8916-samsung-serranove.dts is done.

File: arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	gpio_keys_default: gpio-keys-default {
+		pins = "gpio107", "gpio109";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-pull-up;
+	};	
+};
+
+&pm8916_resin {
```
Comment (TravMurav) [inline arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts:227]: While usually nodes are sorted strictly alphabetically, right now msm8916 devices place smd_rpm_regulators and msmgpio last in the file. Let's follow that for now.

File: arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+	};
+
+	l18 {
+		regulator-min-microvolt = <2700000>;
+		regulator-max-microvolt = <2700000>;
+	};
+};
+
+&msmgpio {
+	muic_i2c_default: muic-i2c-default {
```
Comment (TravMurav) [inline arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts:202]: I don't see the user for this.

**Review by TravMurav — COMMENTED** [review]
You will also need to squash the commits and fix the checkpatch warnings/errors. Also unsure if you're going to use gprime common files later

File: arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+	muic_int_default: muic-int-default {
+		pins = "gpio12";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	gpio_keys_default: gpio-keys-default {
```
Comment (stephan-gh) [inline arch/arm/boot/dts/qcom-msm8916-samsung-heatqlte.dts:218]: and within &msmgpio it should be alphabetically ordered.

**Comment by garethppls** [comment]
I’m going to switch branch on this. I’ve got most of the way there on the touch driver. Will present it when it is cleaner.

**Pull request closed without merging**